.github/workflows/GNU.yml

```diff
@@ -38,42 +38,63 @@ jobs:
-        sed -i -e '/tests\/cp\/file-perm-race.sh/ D' Makefile # issue #1710
-        sed -i -e '/tests\/cp\/special-f.sh/ D' Makefile # issue #1710
-        sed -i -e '/tests\/mv\/mv-special-1.sh/ D' Makefile # issue #1710
-        sed -i -e '/tests\/dd\/stats.sh/ D' Makefile # issue #1710
-        sed -i -e '/tests\/cp\/existing-perm-race.sh/ D' Makefile # hangs, cp doesn't implement --copy-contents
-        # Remove tests that rely on seq with large numbers. See issue #1703
-        sed -i -e '/tests\/misc\/seq.pl/ D' \
-               -e '/tests\/misc\/seq-precision.sh/D' \
-               Makefile
+        sed -i -e 's|^seq |/usr/bin/seq |' -e "s|sha1sum |$(which sha1sum) |" tests/factor/t*sh
+
         # Remove tests checking for --version & --help
         # Not really interesting for us and logs are too big
         sed -i -e '/tests\/misc\/invalid-opt.pl/ D' \
                -e '/tests\/misc\/help-version.sh/ D' \
                -e '/tests\/misc\/help-version-getopt.sh/ D' \
                Makefile
-        sed -i -e '/tests\/tail-2\/pid.sh/ D' Makefile # hangs on github, tail doesn't support -f
-        sed -i -e'/incompat4/ D' -e"/options '-co' are incompatible/ d" tests/misc/sort.pl # Sort doesn't correctly check for incompatible options, waits for input
+
+        # Use the system coreutils where the test fails due to error in a util that is not the one being tested
+        sed -i -e "s|stat|$(which stat)|" tests/chgrp/basic.sh tests/cp/existing-perm-dir.sh tests/touch/60-seconds.sh tests/misc/sort-compress-proc.sh tests/touch/60-seconds.sh
+        sed -i -e "s|ls -|$(which ls) -|" tests/chgrp/posix-H.sh tests/chown/deref.sh tests/cp/same-file.sh tests/misc/mknod.sh tests/mv/part-symlink.sh
+        sed -i -e "s|mkdir |$(which mkdir) |" tests/cp/existing-perm-dir.sh tests/rm/empty-inacc.sh
+        sed -i -e "s|timeout |$(which timeout) |" tests/cp/parent-perm-race.sh tests/ls/infloop.sh tests/misc/sort-exit-early.sh tests/misc/sort-NaN-infloop.sh tests/misc/uniq-perf.sh tests/tail-2/inotify-only-regular.sh tests/tail-2/pipe-f2.sh tests/tail-2/retry.sh tests/tail-2/symlink.sh tests/tail-2/wait.sh tests/tail-2/pid.sh tests/dd/stats.sh
+        sed -i -e "s| timeout | $(which timeout) |" tests/tail-2/inotify-rotate.sh # Don't break the function called 'grep_timeout'
+        sed -i -e "s|chmod |$(which chmod) |" tests/du/inacc-dir.sh tests/mkdir/p-3.sh
+        sed -i -e "s|sort |$(which sort) |" tests/ls/hyperlink.sh tests/misc/test-N.sh
+        sed -i -e "s|split |$(which split) |" tests/misc/factor-parallel.sh
+        sed -i -e "s|truncate |$(which truncate) |" tests/split/fail.sh
+
+        #Add specific timeout to tests that currently hang to limit time spent waiting
+        sed -i -e "s|seq \\$|$(which timeout) 0.1 seq \$|" tests/misc/seq-precision.sh tests/misc/seq-long-double.sh
+        sed -i -e "s|cat |$(which timeout) 0.1 cat |" tests/misc/cat-self.sh
```

Comment:
The `-e` options are not needed here. The `which` commands are also not needed, since the GNU coreutils should always be in `/usr/bin/`.